packages/diff/src/Diff.ts

```diff
@@ -0,0 +1,128 @@
+import { Node, Path } from '@finnair/path';
+
+export const defaultDiffFilter = (_path: Path, value: any) => value !== undefined;
+
+export interface DiffConfig {
+  readonly filter?: DiffFilter;
+  readonly isPrimitive?: (value: any) => boolean; 
+  readonly isEqual?: (a: any, b: any) => boolean;
+  readonly includeObjects?: boolean;
+}
+
+export interface DiffFilter {
+  (path: Path, value: any): boolean;
+}
+
+export interface Change {
+  readonly path: Path;
+  readonly oldValue?: any;
+  readonly newValue?: any;
+}
+
+const OBJECT = Object.freeze({});
+const ARRAY = Object.freeze([]);
+
+const primitiveTypes: any = {
+  'boolean': true,
+  'number': true,
+  'string': true,
+  'bigint': true,
+  'symbol': true,
+};
+
+function isPrimitive(value: any) {
+  return value === null || value === undefined || !!primitiveTypes[typeof value]; 
+}
+
+export class Diff {
+
```

Comment:
Empty line could be removed (I got nothing else to complain about, so now I'm picking on empty lines 🤷‍♂️)